nemo_text_processing/text_normalization/hi/taggers/telephone.py

```diff
@@ -0,0 +1,227 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import pynini
+from pynini.lib import pynutil
+
+from nemo_text_processing.text_normalization.hi.graph_utils import (
+    NEMO_CHAR,
+    NEMO_DIGIT,
+    NEMO_HI_DIGIT,
+    NEMO_SPACE,
+    NEMO_WHITE_SPACE,
+    GraphFst,
+    delete_space,
+    insert_space,
+)
+from nemo_text_processing.text_normalization.hi.utils import get_abs_path
+
+delete_zero = pynutil.delete(pynini.union("0", "०"))
+delete_zero_optional = pynini.closure(delete_zero, 0, 1)
+insert_shunya = pynutil.insert('शून्य') + insert_space
+
+# Load the number mappings from the TSV file
+digit_to_word = pynini.string_file(get_abs_path("data/telephone/number.tsv"))
+digits = pynini.string_file(get_abs_path("data/numbers/digit.tsv"))
+zero = pynini.string_file(get_abs_path("data/numbers/zero.tsv"))
+mobile_context = pynini.string_file(get_abs_path("data/telephone/mobile_context.tsv"))
+landline_context = pynini.string_file(get_abs_path("data/telephone/landline_context.tsv"))
+credit_context = pynini.string_file(get_abs_path("data/telephone/credit_context.tsv"))
+pincode_context = pynini.string_file(get_abs_path("data/telephone/pincode_context.tsv"))
+
+
+def generate_mobile(context_keywords):
+    context_before, context_after = get_context(context_keywords)
+
+    allowed_digits = pynini.union("६", "७", "८", "९", "6", "7", "8", "9")
+
+    # Filter cardinals to only include allowed digits
+    mobile_start_digit = allowed_digits @ digits | allowed_digits @ digit_to_word
+
+    country_code_digits = pynini.closure((digit_to_word | digits | zero) + insert_space, 1, 3)
```

Comment:
you repeat (digit_to_word | digits | zero) a lot in this file. Just define once, optimize, then reuse.